File: CAGradientLayer Example ObjC/CAGradientLayer Example ObjC/ViewController.m
//
//  ViewController.m
//  CAGradientLayer Example ObjC
//
//  Created by Kyle Howells on 14/06/2020.
//  Copyright © 2020 Kyle Howells. All rights reserved.
//

#import "ViewController.h"
#import "KHGradientView.h"

@interface ViewController (){
	KHGradientView *linGradientView;
	KHGradientView *radGradientView;
	KHGradientView *angGradientView;
}
@end

@implementation ViewController

-(BOOL)prefersStatusBarHidden{
	return YES;
}

- (void)viewDidLoad {
	[super viewDidLoad];
	
	//
	// Linear Gradient
	//
	linGradientView = [[KHGradientView alloc] init];
	linGradientView.layer.type = kCAGradientLayerAxial;
	// The colors
	linGradientView.layer.colors =
	@[
		(id)[UIColor colorWithRed: 48.0/255.0 green: 35.0/255.0 blue: 174.0/255.0 alpha: 1.0].CGColor,
		(id)[UIColor colorWithRed: 200.0/255.0 green: 109.0/255.0 blue: 215.0/255.0 alpha: 1.0].CGColor
	];
	// Start and end position (x:0 left - x:1 right) (y:0 top - y:1 bottom)
	linGradientView.layer.startPoint = CGPointZero;
	linGradientView.layer.endPoint = CGPointMake(1, 1);
	[self.view addSubview:linGradientView];
	
	
	//
	// Radial Graient
	//
	radGradientView = [[KHGradientView alloc] init];
	radGradientView.layer.type = kCAGradientLayerRadial;
	radGradientView.layer.colors =
	@[
		(id)[UIColor colorWithRed: 0.0/255.0 green: 101.0/255.0 blue: 255.0/255.0 alpha: 1.0].CGColor,
		(id)[UIColor colorWithRed: 0.0/255.0 green: 40.0/255.0 blue: 101.0/255.0 alpha: 1.0].CGColor
	];
	// Start point of first color
	radGradientView.layer.startPoint = CGPointMake(0.5, 0.5);
	// End points of the x and y edges of the second color circle
	radGradientView.layer.endPoint = CGPointMake(0, 1);
	[self.view addSubview:radGradientView];
	
	
	//
	// Angluar Gradient
	//
	angGradientView = [[KHGradientView alloc] init];
	angGradientView.layer.type = kCAGradientLayerConic;
	angGradientView.layer.colors =
	@[
		(id)[UIColor colorWithRed: 245.0/255.0 green: 81.0/255.0 blue: 95.0/255.0 alpha: 1.0].CGColor,
		(id)[UIColor colorWithRed: 159.0/255.0 green: 4.0/255.0 blue: 27.0/255.0 alpha: 1.0].CGColor
	];
	// The center point the gradient spins around
	angGradientView.layer.startPoint = CGPointMake(0.5, 0.5);
	// The point in the view for the point it is drawn pointing towards
	angGradientView.layer.endPoint = CGPointMake(1, 0.5);
	[self.view addSubview:angGradientView];
}


- (void)viewDidLayoutSubviews {
	[super viewDidLayoutSubviews];
	const CGSize size = self.view.bounds.size;
	
	if (size.height > size.width)
	{
		// Vertical layout
		CGFloat height = size.height/3;
		
		linGradientView.frame = ({
			CGRect frame = CGRectZero;
			frame.size.width = size.width;
			frame.size.height = height;
			frame;
		});
		
		radGradientView.frame = ({
			CGRect frame = CGRectZero;
			frame.origin.y = height;
			frame.size.width = size.width;
			frame.size.height = height;
			frame;
		});
		
		angGradientView.frame = ({
			CGRect frame = CGRectZero;
			frame.origin.y = (size.height - height);
			frame.size.width = size.width;
			frame.size.height = height;
			frame;
		});
	}
	else
	{
		// Horizontal layout
		CGFloat width = size.width/3;
		
		linGradientView.frame = ({
			CGRect frame = CGRectZero;
			frame.size.width = width;
			frame.size.height = size.height;
			frame;
		});
		
		radGradientView.frame = ({
			CGRect frame = CGRectZero;
			frame.origin.x = width;
			frame.size.width = width;
			frame.size.height = size.height;
			frame;
		});
		
		angGradientView.frame = ({
			CGRect frame = CGRectZero;
			frame.origin.x = (size.width - width);
			frame.size.width = width;
			frame.size.height = size.height;
			frame;
		});
	}
}

@end

File: CAGradientLayer Example Swift/CAGradientLayer Example Swift/ViewController.swift
//
//  ViewController.swift
//  CAGradientLayer Example Swift
//
//  Created by Kyle Howells on 15/06/2020.
//  Copyright © 2020 Kyle Howells. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

	//
	// Linear Gradient
	//
	lazy var linGradientView:KHGradientView = {
		let v = KHGradientView()
		v.gradientLayer.type = CAGradientLayerType.axial
		v.gradientLayer.colors =
		[
			UIColor(red: 48.0/255.0, green: 35.0/255.0, blue: 174.0/255.0, alpha: 1.0).cgColor,
			UIColor(red: 200.0/255.0, green: 109.0/255.0, blue: 215.0/255.0, alpha: 1.0).cgColor
		]
		v.gradientLayer.startPoint = CGPoint.zero
		v.gradientLayer.endPoint = CGPoint(x: 1, y: 1)
		return v
	}()
	
	
	//
	// Radial Gradient
	//
	lazy var radGradientView:KHGradientView = {
		let v = KHGradientView()
		v.gradientLayer.type = CAGradientLayerType.radial
		v.gradientLayer.colors =
		[
			UIColor(red: 0.0/255.0, green: 101.0/255.0, blue: 255.0/255.0, alpha: 1.0).cgColor,
			UIColor(red: 0.0/255.0, green: 40.0/255.0, blue: 101.0/255.0, alpha: 1.0).cgColor
		]
		v.gradientLayer.startPoint = CGPoint(x: 0.5, y: 0.5)
		v.gradientLayer.endPoint = CGPoint(x: 0, y: 1)
		return v
	}()
	
	
	//
	// Angluar Gradient
	//
	lazy var angGradientView:KHGradientView = {
		let v = KHGradientView()
		v.gradientLayer.type = CAGradientLayerType.conic
		v.gradientLayer.colors =
		[
			UIColor(red: 245.0/255.0, green: 81.0/255.0, blue: 95.0/255.0, alpha: 1.0).cgColor,
			UIColor(red: 159.0/255.0, green: 4.0/255.0, blue: 27.0/255.0, alpha: 1.0).cgColor
		]
		v.gradientLayer.startPoint = CGPoint(x: 0.5, y: 0.5)
		v.gradientLayer.endPoint = CGPoint(x: 1, y: 0.5)
		return v
	}()
	
	override func viewDidLoad() {
		super.viewDidLoad()
		// Do any additional setup after loading the view.
		
		self.view.addSubview(self.linGradientView)
		self.view.addSubview(self.radGradientView)
		self.view.addSubview(self.angGradientView)
	}

	override var prefersStatusBarHidden: Bool {
		return true
	}

	override func viewDidLayoutSubviews() {
		super.viewDidLayoutSubviews()
		let size = self.view.bounds.size
		
		if size.height > size.width {
			// Vertical layout
			
			let height = size.height / 3
			
			self.linGradientView.frame = {
				var frame = CGRect()
				frame.size.height = height
				frame.size.width = size.width
				return frame
			}()
			
			self.radGradientView.frame = {
				var frame = CGRect()
				frame.origin.y = height
				frame.size.height = height
				frame.size.width = size.width
				return frame
			}()
			
			self.angGradientView.frame = {
				var frame = CGRect()
				frame.origin.y = size.height - height
				frame.size.height = height
				frame.size.width = size.width
				return frame
			}()
			
		}
		else {
			// Horizontal layout
			
			let width = size.width / 3
			
			self.linGradientView.frame = {
				var frame = CGRect()
				frame.size.height = size.height
				frame.size.width = width
				return frame
			}()
			
			self.radGradientView.frame = {
				var frame = CGRect()
				frame.origin.x = width
				frame.size.height = size.height
				frame.size.width = width
				return frame
			}()
			
			self.angGradientView.frame = {
				var frame = CGRect()
				frame.origin.x = size.width - width
				frame.size.height = size.height
				frame.size.width = width
				return frame
			}()
		}

	}
}
